src/backend/distributed/transaction/transaction_management.c

```diff
@@ -186,6 +201,17 @@ InCoordinatedTransaction(void)
 }
 
 
+/*
+ * Returns true if a distributed forceDelegation function is executed in
+ * the current transaction.
+ */
+bool
+GetInForceDelegatedFuncExecution()
+{
+	return ShouldAllowRestricted2PC;
```

Comment:
The variable and function name seem weirdly inconsistent.

Reply:
I know how this happened. Earlier this function was check for AllowedDistributionColumnValue != NULL, and we were resetting it at transaction boundaries, but later we changed it to reset this value after each statement, so we came up with the new flag with a transaction scope. Changed to GetShouldAllowRestricted2PC()